### PR TITLE
(old) drag and drop functionality for tasks in the Project view

### PR DESCRIPTION
**Pull request by azangru**
NOT FOR MERGE

This is obsoleted by PR https://github.com/the-ninja-turtles/turtle/pull/77

Keeping this as a source of comments

### CODE REVIEW

File: frontend/src/components/project/task.jsx [hunk 1]
```diff
@@ -1,19 +1,74 @@
...
+    const draggedId = monitor.getItem().id;
+    if (draggedId !== props.id) {
+      ProjectActions.updateTaskRankLocally({
+        draggedTaskId: draggedId,
+        targetTaskId: props.id
+      });
+    }
+  },
+
+  // on dropping a task card on another task card, a message should be sent to the server
```
Comment (azangru) [inline frontend/src/components/project/task.jsx:25]: I am an idiot! I need to check whether I am dropping a task on itself (as will happen after tasks have swapped during hover) here, not in the ProjectStore as I am doing now. An action should be fired only if a task is dropped on another task and not on itself. I'll fix that.

File: frontend/src/components/project/project.jsx [hunk 2]
```diff
@@ -1,32 +1,86 @@
...
+          <NextSprintInfo
+            sprint={this.state.nextSprint}
+            tasks={this.state.tasksForNextSprint}
+          />
+          <Droparea tasks={this.state.tasksForNextSprint} />
+          <hr />
+          <Backlog tasks={this.state.tasksInBacklog} />
+        </div>
+      );
+    }
```
Comment (azangru) [inline frontend/src/components/project/project.jsx:82]: I wonder whether there is a more concise way to write these if-else statements in JSX. This way looks repetitive.

Comment (wezleytsai) [inline frontend/src/components/project/project.jsx:82]: You can nest the if/else like so, but you need to wrap it in an immediately invoked function.

```
return (
  <div className='project-view'>
    {(() => {
      if (this.state.currentSprint) {
        return (
          <CurrentSprintInfo
            sprint={this.state.currentSprint}
            tasks={this.state.tasksInCurrentSprint}
          />
        );
      } else {
        return (
          <NextSprintInfo
            sprint={this.state.nextSprint}
            tasks={this.state.tasksForNextSprint}
          />
        );
      }
    })()}
    <Droparea tasks={this.state.tasksForNextSprint} />
    <hr />
    <Backlog tasks={this.state.tasksInBacklog} />
  </div>
);
```

Comment (wezleytsai) [inline frontend/src/components/project/project.jsx:82]: Or you can do something like this:

```
let sprint;
if (this.state.currentSprint) {
  sprint = (
    <CurrentSprintInfo
      sprint={this.state.currentSprint}
      tasks={this.state.tasksInCurrentSprint}
    />
  );
} else {
  sprint = (
    <NextSprintInfo
      sprint={this.state.nextSprint}
      tasks={this.state.tasksForNextSprint}
    />
  );
}

return (
  <div className='project-view'>
    {
      sprint
    }
    <Droparea tasks={this.state.tasksForNextSprint} />
    <hr />
    <Backlog tasks={this.state.tasksInBacklog} />
  </div>
);
```

**Pull request closed without merging**